src/java/org/apache/cassandra/io/util/ThreadLocalReadAheadBuffer.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.io.util;
+
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.netty.util.concurrent.FastThreadLocal;
+import org.apache.cassandra.io.compress.BufferType;
+import org.apache.cassandra.io.compress.CorruptBlockException;
+import org.apache.cassandra.io.sstable.CorruptSSTableException;
+
+public final class ThreadLocalReadAheadBuffer
+{
+
+    private final ChannelProxy channel;
+
+    private final BufferType bufferType;
+
+    private static final Map<String, FastThreadLocal<ByteBuffer>> blockBufferHolders = new ConcurrentHashMap<>();
+
+    private static final Map<String, FastThreadLocal<Integer>> storedBlockNos = new ConcurrentHashMap<>();
+
+    private final FastThreadLocal<ByteBuffer> blockBufferHolder;
+
+    private final FastThreadLocal<Integer> storedBlockNo;
```

Comment:
bike-shedding: could do something like `currentBlock` to indicate this is kind of a pointer/cursor, but not hard to figure out what it is either way after looking for 2 minutes

Reply:
Will rename